src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -293,209 +306,234 @@ private async Task<SearchResult> SearchImpl(SqlSearchOptions sqlSearchOptions, S
+                    if (exportTimeTravel)
                     {
-                        await reader.ReadAsync(cancellationToken);
-                        long count = reader.GetInt64(0);
-                        if (count > int.MaxValue)
-                        {
-                            _requestContextAccessor.RequestContext.BundleIssues.Add(
-                                new OperationOutcomeIssue(
-                                    OperationOutcomeConstants.IssueSeverity.Error,
-                                    OperationOutcomeConstants.IssueType.NotSupported,
-                                    string.Format(Core.Resources.SearchCountResultsExceedLimit, count, int.MaxValue)));
+                        PopulateSqlCommandFromQueryHints(clonedSearchOptions, sqlCommand);
+                        sqlCommand.CommandTimeout = 1200; // set to 20 minutes, as dataset is usually large
+                    }
+                    else
+                    {
+                        var stringBuilder = new IndentedStringBuilder(new StringBuilder());
 
-                            throw new InvalidSearchOperationException(string.Format(Core.Resources.SearchCountResultsExceedLimit, count, int.MaxValue));
-                        }
+                        EnableTimeAndIoMessageLogging(stringBuilder, connection);
 
-                        var searchResult = new SearchResult((int)count, clonedSearchOptions.UnsupportedSearchParams);
+                        var queryGenerator = new SqlQueryGenerator(
+                            stringBuilder,
+                            new HashingSqlQueryParameterManager(new SqlQueryParameterManager(sqlCommand.Parameters)),
+                            _model,
+                            searchType,
+                            _schemaInformation,
+                            currentSearchParameterHash);
 
-                        // call NextResultAsync to get the info messages
-                        await reader.NextResultAsync(cancellationToken);
+                        expression.AcceptVisitor(queryGenerator, clonedSearchOptions);
 
-                        return searchResult;
-                    }
+                        SqlCommandSimplifier.RemoveRedundantParameters(stringBuilder, sqlCommand.Parameters, _logger);
 
-                    var resources = new List<SearchResultEntry>(sqlSearchOptions.MaxItemCount);
-                    short? newContinuationType = null;
-                    long? newContinuationId = null;
-                    bool moreResults = false;
-                    int matchCount = 0;
+#pragma warning disable CA2100 // Review SQL queries for security vulnerabilities
+                        sqlCommand.CommandText = stringBuilder.ToString();
+#pragma warning restore CA2100 // Review SQL queries for security vulnerabilities
```

Comment:
>Review SQL queries for security vulnerabilities [](http://example.com/codeflow?start=34&length=47)

Add "no direct user input" in the comments

Reply:
Done